contribs/gnofaucet/gh_fetcher.go

```diff
@@ -0,0 +1,373 @@
+	for org, names := range f.repos {
+		for _, n := range names {
+			f.logger.Info("fetching events for repo", zap.String("org", org), zap.String("repo", n))
+			pipe := f.redisClient.TxPipeline()
+			if !f.iterateEvents(ctx, pipe, org, n) {
+				f.logger.Error("error fetching events. Aborting the pipeline", zap.String("org", org), zap.String("repo", n))
+				continue
+			}
+			_, err := pipe.Exec(ctx)
+			if err != nil {
+				f.logger.Error("error executing redis pipeline", zap.String("org", org), zap.String("repo", n), zap.Error(err))
+				return err
+			} else {
+				f.logger.Info("events correctly iterated", zap.String("org", org), zap.String("repo", n))
+			}
+		}
+	}
+
+	return nil
+}
+
+func (f *GHFetcher) fetchHistory(ctx context.Context) error {
+	for org, names := range f.repos {
+		for _, n := range names {
+			f.logger.Info("Fetching history for repo", zap.String("org", org), zap.String("repo", n))
+			pipe := f.redisClient.TxPipeline()
+
+			if !f.iterateIssues(ctx, pipe, org, n) {
+				return fmt.Errorf("error iterating issues")
+			}
+			if !f.iteratePullRequests(ctx, pipe, org, n) {
+				return fmt.Errorf("error iterating pull requests")
+			}
+
+			_, err := pipe.Exec(ctx)
+			if err != nil {
+				return fmt.Errorf("error executing redis pipeline: %w", err)
+			} else {
+				f.logger.Info("history saved", zap.String("org", org), zap.String("repo", n))
+			}
+		}
+	}
+
+	return nil
+}
+
+func (f *GHFetcher) processIssue(ctx context.Context, pipe redis.Pipeliner, org, repo string, issue *github.Issue) {
+	if issue.User == nil {
+		return
+	}
+	u := issue.User.Login
+	if u == nil {
+		return
+	}
+
+	f.logger.Info("processing issue", zap.String("user", *u), zap.String("org", org), zap.String("repo", repo), zap.Any("issue", issue.Number))
+
+	if err := pipe.Incr(ctx, issueCountKey(*u)).Err(); err != nil {
+		f.logger.Error("error adding issue to the count", zap.String("user", *u), zap.String("org", org), zap.String("repo", repo), zap.Error(err))
+	}
+}
+
+func (f *GHFetcher) processPullRequest(ctx context.Context, pipe redis.Pipeliner, org, repo string, pr PullRequest) {
+	u := pr.Author()
+	if u == "" {
+		return
+	}
+
+	f.logger.Info("processing pr", zap.String("user", u), zap.String("org", org), zap.String("repo", repo), zap.Any("pr", pr.Number()))
+
+	if err := pipe.Incr(ctx, prCountKey(u)).Err(); err != nil {
+		f.logger.Error("error adding prs to the count", zap.String("user", u), zap.String("org", org), zap.String("repo", repo), zap.Error(err))
+		return
+	}
+	// if PR is not merged, do not count the commits
+	if pr.State() != string(PullRequestStateMerged) {
+		return
+	}
+	if pr.CommitsCount() != 0 {
+		if err := pipe.IncrBy(ctx, commitCountKey(u), int64(pr.CommitsCount())).Err(); err != nil {
+			f.logger.Error("error adding commits to the count", zap.String("user", u), zap.String("org", org), zap.String("repo", repo), zap.Error(err))
+		}
+	}
+
+	for _, r := range pr.Reviews() {
```

Comment:
We should be careful with this, to make sure people are not spamming comments / reviews. I'm not sure if this should even be counted (even for specific user groups), because it's an easily game-able metric

Merged PRs are the bread and butter of what moves contributions

Reply:
Whatever is fine for me, I was just mimicking the behavior on gnolove, where PR reviews are adding points.

Reply:
Also FYI, there can be several reviews for a single user on the same PR. We had to modify the `github-bot` so it handles this kind of case. For example, [on this PR](https://github.com/gnolang/gno/pull/3678) I have 3 reviews: 2 `CHANGES_REQUESTED` (due to a misclick) + the `APPROVED`.